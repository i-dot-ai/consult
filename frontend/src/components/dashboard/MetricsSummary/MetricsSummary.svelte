<script lang="ts">
    import { type Component } from "svelte";

    import MaterialIcon from "../../MaterialIcon.svelte";
    import Description from "../../svg/material/Description.svelte";
    import Group from "../../svg/material/Group.svelte";
    import Shield from "../../svg/material/Shield.svelte";


    interface Props {
        questionCount: number;
        responseCount: number;
        demoCount: number;
    }

    let {
        questionCount = 0,
        responseCount = 0,
        demoCount = 0,
    }: Props = $props();
</script>

{#snippet row(text: string, count: number, icon: Component)}
    <div class="flex items-center justify-between my-2">
        <div class="flex items-center">
            <MaterialIcon color="fill-neutral-500">
                <svelte:component this={icon} />
            </MaterialIcon>

            <span class="ml-2 text-sm text-neutral-700">
                {text}
            </span>
        </div>

        <span class="text-xl font-bold">
            {count.toLocaleString()}
        </span>
    </div>
{/snippet}

{@render row( "Responses", questionCount, Description )}
{@render row( "Questions", responseCount, Group )}
{@render row( "Demographics",  demoCount, Shield )}