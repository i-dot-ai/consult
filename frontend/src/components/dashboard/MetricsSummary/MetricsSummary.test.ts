import { afterEach, describe, expect, it } from "vitest";
import { render, cleanup } from "@testing-library/svelte";

import MetricsSummary, { type Props } from "./MetricsSummary.svelte";

describe("MetricsSummary", () => {
  afterEach(() => cleanup());

  it("should render data", () => {
    const TEST_DATA: Props = {
      questionCount: 10,
      responseCount: 20,
      demoCount: 30,
    };

    const { getByText } = render(MetricsSummary, {
      questionCount: TEST_DATA.questionCount,
      responseCount: TEST_DATA.responseCount,
      demoCount: TEST_DATA.demoCount,
    });

    expect(getByText("Responses"));
    expect(getByText(TEST_DATA.responseCount));

    expect(getByText("Questions"));
    expect(getByText(TEST_DATA.questionCount));

    expect(getByText("Demographics"));
    expect(getByText(TEST_DATA.demoCount));
  });
});
