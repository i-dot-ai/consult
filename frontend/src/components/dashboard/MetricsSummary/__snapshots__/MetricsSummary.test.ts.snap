// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MetricsSummary > should render data 1`] = `
<div>
  <!---->
  <div
    class="w-full flex items-center justify-between my-2 gap-4"
  >
    <div
      class="flex items-center"
    >
      <svg
        class="fill-neutral-500"
        height="1rem"
        role="img"
        viewBox="0 -960 960 960"
        width="1rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <!---->
        <path
          d="M40-160v-112q0-34 17.5-62.5T104-378q62-31 126-46.5T360-440q66 0 130 15.5T616-378q29 15 46.5 43.5T680-272v112H40Zm720 0v-120q0-44-24.5-84.5T666-434q51 6 96 20.5t84 35.5q36 20 55 44.5t19 53.5v120H760ZM360-480q-66 0-113-47t-47-113q0-66 47-113t113-47q66 0 113 47t47 113q0 66-47 113t-113 47Zm400-160q0 66-47 113t-113 47q-11 0-28-2.5t-28-5.5q27-32 41.5-71t14.5-81q0-42-14.5-81T544-792q14-5 28-6.5t28-1.5q66 0 113 47t47 113ZM120-240h480v-32q0-11-5.5-20T580-306q-54-27-109-40.5T360-360q-56 0-111 13.5T140-306q-9 5-14.5 14t-5.5 20v32Zm240-320q33 0 56.5-23.5T440-640q0-33-23.5-56.5T360-720q-33 0-56.5 23.5T280-640q0 33 23.5 56.5T360-560Zm0 320Zm0-400Z"
        />
        
        <!---->
      </svg>
      <!---->
       
      <span
        class="ml-2 text-sm text-neutral-700"
      >
        Questions
      </span>
    </div>
     
    <span
      class="text-xl font-bold"
    >
      10
    </span>
  </div>
  <!---->
   
  <div
    class="w-full flex items-center justify-between my-2 gap-4"
  >
    <div
      class="flex items-center"
    >
      <svg
        class="fill-neutral-500"
        height="1rem"
        role="img"
        viewBox="0 -960 960 960"
        width="1rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <!---->
        <path
          d="M320-240h320v-80H320v80Zm0-160h320v-80H320v80ZM240-80q-33 0-56.5-23.5T160-160v-640q0-33 23.5-56.5T240-880h320l240 240v480q0 33-23.5 56.5T720-80H240Zm280-520v-200H240v640h480v-440H520ZM240-800v200-200 640-640Z"
          xmlns="http://www.w3.org/2000/svg"
        />
        
        <!---->
      </svg>
      <!---->
       
      <span
        class="ml-2 text-sm text-neutral-700"
      >
        Responses
      </span>
    </div>
     
    <span
      class="text-xl font-bold"
    >
      20
    </span>
  </div>
  <!---->
   
  <div
    class="w-full flex items-center justify-between my-2 gap-4"
  >
    <div
      class="flex items-center"
    >
      <svg
        class="fill-neutral-500"
        height="1rem"
        role="img"
        viewBox="0 -960 960 960"
        width="1rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <!---->
        <path
          d="M480-80q-139-35-229.5-159.5T160-516v-244l320-120 320 120v244q0 152-90.5 276.5T480-80Zm0-84q104-33 172-132t68-220v-189l-240-90-240 90v189q0 121 68 220t172 132Zm0-316Z"
          xmlns="http://www.w3.org/2000/svg"
        />
        
        <!---->
      </svg>
      <!---->
       
      <span
        class="ml-2 text-sm text-neutral-700"
      >
        Demographics
      </span>
    </div>
     
    <span
      class="text-xl font-bold"
    >
      30
    </span>
  </div>
  <!---->
  
</div>
`;
